### PR TITLE
link-click-tracking: event delegation

### PR DESCRIPTION
**Pull request by jethron**
There are several API changes included:

- Global event listeners are now used instead of per-element listeners
- `refreshLinkClickTracking` is now deprecated and does nothing as it should no longer be required (but still exists for API compatibility)
- There is now a `disableLinkClickTracking` method to balance `enableLinkClickTracking`
- `trackLinkClick` can now optionally be passed a link element directly, rather than requiring manual construction of the event payload

See PE-4138

### CODE REVIEW

File: plugins/browser-plugin-link-click-tracking/src/index.ts [hunk 1]
```diff
@@ -83,178 +99,200 @@ export interface LinkClickTrackingConfiguration {
...
-    sourceElement = parentElement;
+  // Ignore script pseudo-protocol links
+  if (!ELEMENT_PROTOCOL_FILTER.test(sourceHref)) {
+    elementId = anchorElement.id;
+    elementClasses = getCssClasses(anchorElement);
+    elementTarget = anchorElement.target;
+    elementContent = includeContent ? anchorElement.innerHTML : undefined;
+
+    // decodeUrl %xx
+    // NOTE(jethron): Why do we do this?
```
Comment (matus-tomlein) [inline plugins/browser-plugin-link-click-tracking/src/index.ts:211]: I think it would be a good idea to drop the `unescape` or replace it with `decodeURI`. Based [on the docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape), they don't recommend using it.

I assume what we want to do here is just to make the URI look nicer so that a URL like:
`https://snowplow.io/?x=a%20b%20c`
will turn into:
`https://snowplow.io/?x=a b c`

I am not convinced it's a good idea, because we don't seem to do it for the URL property in page views, which creates an inconsistency in how the URL is represented.

I think that as part of the breaking changes of the v4, we can also drop this. What do you think?

File: plugins/browser-plugin-link-click-tracking/test/events.test.ts [hunk 2]
```diff
@@ -36,17 +36,15 @@ const getUEEvents = F.compose(F.filter(F.compose(F.eq('ue'), F.get('evt.e'))));
...
+    from: (a: any, n: number = 0) =>
+      F.nth(
+        n,
+        F.compose(F.filter(F.compose(F.eq(schema), F.get('schema'))), F.flatten, extractEventProperties, getUEEvents)(a)
+      ),
   };
 };
 
-describe('AdTrackingPlugin', () => {
+describe('LinkClickTrackingPlugin', () => {
```
Comment (matus-tomlein) [inline plugins/browser-plugin-link-click-tracking/test/events.test.ts:47]: The link click tracking plugin has one of the lowest unit test coverage in the tracker (around 18% last time we checked it). This is a bit increasing the scope of the PR, but would you consider adding a few unit tests to cover more of the event listening? Considering how flaky the integration tests are, we wanted to move more of the testing to the unit tests. Peter has done some unit tests in the cross domain plugin that use jest to mock links and also test link clicks ([see here](https://github.com/snowplow/snowplow-javascript-tracker/blob/master/libraries/browser-tracker-core/test/tracker/cross_domain.test.ts#L53)). Maybe we could do something similar here if you have the capacity left?

**Comment by matus-tomlein** [comment]
I have just rebased the `release/4.0.0` branch on top of master since it was quite a bit behind which resulted in a few conflicts here. Sorry about that, hope it won't be too much trouble to rebase!

File: plugins/browser-plugin-link-click-tracking/src/index.ts (hunk 1 above)
Comment (jethron) [reply plugins/browser-plugin-link-click-tracking/src/index.ts:211]: Yeah I agree; it's a lossy transform and would be hard to parse after the fact for any non-human-legibility use case. I'll remove this.

File: plugins/browser-plugin-link-click-tracking/test/events.test.ts (hunk 2 above)
Comment (matus-tomlein) [reply plugins/browser-plugin-link-click-tracking/test/events.test.ts:47]: Very nice tests, thanks! I think @igneel64 used [istanbul.js](https://istanbul.js.org) to calculate the code coverage, there is [a report here](https://www.notion.so/keep-in-the-snow/Trackers-Integration-Testing-overview-Web-focused-cae853ad6ff2452da7275479ec99976f).

**Review by matus-tomlein — APPROVED** [review]
LGTM! Very nice work and seems to be working well!

**Comment by matus-tomlein** [comment]
A sidenote: when you merge the PR, can you please use "squash and merge" or rename the commit to follow the style that our other commits do: https://github.com/snowplow/snowplow-javascript-tracker/commit/2d5b8afc873ee5dcf5b02837a96fd0066e48254e
